CMakeLists.txt

```diff
@@ -236,6 +236,18 @@ find_package(CURL REQUIRED)
 
 find_package(OpenSSL REQUIRED COMPONENTS Crypto SSL)
 
+set(IRODS_USE_LIBSYSTEMD_DOCSTRING "Use libsystemd for service manager status communication")
+if (NOT DEFINED IRODS_USE_LIBSYSTEMD)
+	# if IRODS_USE_LIBSYSTEMD isn't set, check for systemd and set default value based on its presence.
+	find_package(systemd)
+	set(IRODS_USE_LIBSYSTEMD "${SYSTEMD_FOUND}" CACHE BOOL "${IRODS_USE_LIBSYSTEMD_DOCSTRING}")
+else()
+	set(IRODS_USE_LIBSYSTEMD "${IRODS_USE_LIBSYSTEMD}" CACHE BOOL "${IRODS_USE_LIBSYSTEMD_DOCSTRING}")
+	if (IRODS_USE_LIBSYSTEMD)
+		find_package(systemd REQUIRED)
+	endif()
+endif()
```

Comment:
server package needs to add libsystemd as a dependency if `IRODS_USE_LIBSYSTEMD` is true.